scripts/fetch-commit-data

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Obtains fix commit statistics for all bug descriptions."""
+import os
+import typing as t
+
+import git
+from robust import BugDescription, ROBUST
+
+
+def print_skip(bug: BugDescription, reason: str) -> None:
+    print(f"Skipping bug: {bug.id} ({reason})")
+
+
+def process_commit(robust: ROBUST,
+                   bug: BugDescription,
+                   dict_commit: t.Dict[str, t.Any]
+                   ) -> None:
+    if 'stats' in dict_commit:
+        raise ValueError("already has 'stats' field")
+
+    try:
+        repo_url = dict_commit['repo']
+        repo_hash = dict_commit['hash']
+    except KeyError as err:
+        raise ValueError(f"commit is mising field: {err}")
+
+    repo = robust.repo(repo_url)
+    try:
+        fix_commit = repo.commit(repo_hash)
+    except (git.exc.BadName, ValueError):
+        raise ValueError(bug, f"bad fix hash: {repo_hash}")
+
+    commit_stats = fix_commit.stats
+    dict_commit['stats'] = {
+        'total': dict(commit_stats.total),
+        'files': {filename: dict(file_stats)
+                  for filename, file_stats in commit_stats.files.items()}
+    }
+    print(dict_commit)
+
+
+def process_bug(robust: ROBUST, bug: BugDescription) -> None:
+    print(f"Processing bug: {bug.id} [{bug.filename}]")
+
+    try:
+        yaml_commits = bug.yaml['fix']['commits']
+    except KeyError as err:
+        print_skip(bug, f"missing field: {err}")
+        return
+
+    if not isinstance(yaml_commits, list):
```

Comment:
Why not `type(yaml_commits) == list`?

Reply:
Yeah, `type(...)` would be fine here. I'm just in the habit of using `isinstance` since it tends to have the semantics that I want when doing these sorts of checks (i.e., allowing instances of a subclass).

Reply:
The other issue here is that `yaml_commits` is actually a `CommentedSeq` from `ruamel`.

Reply:
Yeah, as I wrote, I was just curious. I'm fine with this as-is.